libs/editor/libs/editor-common/assets/ZSSRichTextEditor.js

```diff
@@ -1138,45 +1144,76 @@ ZSSEditor.removeImage = function(imageNodeIdentifier) {
- *              as soon as it's selected for uploading.  Once the video is successfully uploaded
- *              the application should call replaceLocalVideoWithRemoteVideo().
+ *  @brief      Inserts a placeholder image tag for in-progress video uploads, marked with an identifier.
+ *  @details    The image shown can be the video's poster if available - otherwise the default poster image is used.
+ *              Using an image instead of a video placeholder is a departure from iOS, necessary because the original
+ *              method caused occasional WebView freezes on Android.
+ *              Once the video is successfully uploaded, the application should call replaceLocalVideoWithRemoteVideo().
  *
  *  @param      videoNodeIdentifier     This is a unique ID provided by the caller.  It exists as
  *                                      a mechanism to update the video node with the remote URL
  *                                      when replaceLocalVideoWithRemoteVideo() is called.
  *  @param      posterURL               The URL of a poster image to display while the video is being uploaded.
  */
-ZSSEditor.insertInProgressVideoWithIDUsingPosterImage = function(videoNodeIdentifier, posterURL) {
-    var space = '&nbsp';
+ZSSEditor.insertLocalVideo = function(videoNodeIdentifier, posterURL) {
     var progressIdentifier = this.getVideoProgressIdentifier(videoNodeIdentifier);
     var videoContainerIdentifier = this.getVideoContainerIdentifier(videoNodeIdentifier);
-    var videoContainerStart = '<span id="' + videoContainerIdentifier + '" class="video_container">';
+
+    if (ZSSEditor.androidApiLevel > 18) {
+        var videoContainerClass = 'video_container';
+        var progressElement = '<progress id="' + progressIdentifier + '" value=0 class="wp_media_indicator"'
+                + 'contenteditable="false"></progress>';
+    } else {
+        // Before API 19, the WebView didn't support progress tags. Use an upload overlay instead of a progress bar
+        var videoContainerClass = 'video_container compat';
+        var progressElement = '<span class="upload-overlay" contenteditable="false">' + nativeState.getStringUploading()
+                + '</span><span class="upload-overlay-bg"></span>';
+    }
+
+    var videoContainerStart = '<span id="' + videoContainerIdentifier + '" class="' + videoContainerClass
+            + '" contenteditable="false">';
     var videoContainerEnd = '</span>';
-    var progress = '<progress id="' + progressIdentifier + '" value=0  class="wp_media_indicator"  contenteditable="false"></progress>';
-    var video = '<video data-wpid="' + videoNodeIdentifier + '" webkit-playsinline poster="' + posterURL + '" onclick="" class="uploading"></video>';
-    var html =  space + videoContainerStart + progress + video + videoContainerEnd + space;
-    this.insertHTML(html);
+
+    if (posterURL == '') {
+       posterURL = "wpposter.svg";
```

Comment:
:+1: `posterURL` is never null here (thanks to a call to `StringUtils.notNullStr` on the Java side)